meetings/2023-09-19.md

```diff
@@ -0,0 +1,140 @@
+* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
+* If this is your first time, welcome! please introduce yourself.
+
+
+### Scribes
+* Wouter
+
+
+### Introductions
+* name: text
+
+---
+
+## Topic: Alignment on generalized approach to client-client specs
+
+Context: https://github.com/solid/specification/discussions/554
+
+* eP: Can Tim shortly introduce the new chat spec ([issue](https://github.com/solid/specification/issues/553) / [repo](https://github.com/solid/chat) / [matrix](https://solid.github.io/chat/#channel))?
+* TBL: Basically code we got working on pods is we take a unix directory approach. The same way we tried to follow macos address book (vcards) as basis. Everything else is under this thing. One difference for the chat is that people typically use one address book, while in Solid people will also want to add more than one address book.
+* TBL: There was a 'short chat' idea that worked in a single resource. Ruben suggested to deprecate that in favor of one general 'long chat' approach: a folder with a master file on top; it stores the messages in a dated folder hierachy, like the archive format. So you have a limited amount of chat per day, and then you move on. Then this day resource is appended to the master file. 
+* TBL: Jackson wrote an app that is compatible with Solid and SolidOS. I tried to put all these pieces in it. In particular it got shapes in ShaCL. I now try to integrate with Rebel (?) for shape validation, and integrate with Slack functionalities (store Slack data into my pod).
+* eP: When reading a chat channel, links must be followed, with no assumptions about the URI, with the exception for the hierarchical data structure. Jackson wrote a [comment](https://github.com/solid/specification/issues/553#issuecomment-1688564415) saying: "The main flow lies in the dependenceof the file structure. The entrypoint is index.ttl, but it does not point to where the chat is stored, so developers [must follow this convention]." This also seems to impede on the ...
+* TBL: It would be nice to leave clues, e.g. with a UI template in the chat masterfile indicating the form of the URI. It might indeed be handy to use other structures, but a date here is very common.
+* TBL: The second point is rather philosophically. The main point is that when you create something in the chat (or whatever) you have to be given a folder, then you can do whatever you like with it. Similar to MacOS (?). Example: 
+* `~/Photos/myPhotos.pholLibrary/` is a package and the user doesn't see the substructure of that directory; the app has it to itself.
+* So you might find `../Orignals/20023/09/imh61.jpg` as the structure within that directory.
+* The rule is that an app gets to define all the substructure of a directory.
+* SolidOS uses that for trackers, chat, addressbook, etc
```

Comment:
```suggestion
* eP: When reading a chat channel, links must be followed, with no assumptions about the URI, with an exception for the hierarchical data structure. Jackson wrote a [comment](https://github.com/solid/specification/issues/553#issuecomment-1688564415) saying: "The main flow lies in the dependency on the file structure. The entry point is `index.ttl`, but it does not point to where the chat is stored, so developers [must follow this convention]." This also seems to impede on the ...
* TBL: It would be nice to leave clues, e.g., with a UI template in the chat masterfile, indicating the form of the URI. It might indeed be handy to use other structures, but a date here is very common.
  * ... The second point is rather philosophical. The main point is that when you create something in the chat (or whatever), you have to be given a folder; then you can do whatever you like with it. Similar to macOS (?). Example: 
  * ... `~/Photos/myPhotos.pholLibrary/` is a package and the user doesn't see the substructure of that directory; the app has it to itself.
  * ... So you might find `../Originals/20023/09/imh61.jpg` as the structure within that directory.
  * ... The rule is that an app gets to define the substructure of a directory.
  * ... SolidOS uses that for trackers, chat, address book, etc.
```